charts/opentelemetry-collector/values.yaml

```diff
@@ -574,4 +574,4 @@ networkPolicy:
 # In a future release this setting will be enabled by default.
 # See https://github.com/open-telemetry/opentelemetry-helm-charts/issues/891
 # for more details.
-useGOMEMLIMIT: false
+useGOMEMLIMIT: true
```

Comment:
Enabling this by default does mean that anyone running an install of this chart without resources set just lost the memory ballast extension AND is not getting any value from `GOMEMLIMIT`.  Since the chart no longer has default resources, anyone who installs the default chart will be affected.  NOTES.txt has warning for both not having any resources AND enabling `useGOMEMLIMIT` without resources.


```
❯ helm install testing ./charts/opentelemetry-collector --set mode=deployment
NAME: testing
LAST DEPLOYED: Mon Dec  4 10:53:38 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
[WARNING] No resource limits or requests were set. Consider setter resource requests and limits for your collector(s) via the `resources` field.

[WARNING] "useGOMEMLIMIT" is enabled but memory limits have not been supplied, which means no GOMEMLIMIT env var was configured but the Memory Ballast Extension was removed. It is highly recommended to only use "useGOMEMLIMIT" when memory limits have been set.
```